toolchain/base/llvm_tools.bzl

```
@@ -0,0 +1,149 @@
+# and the library dependency required to use them.
+LLVM_MAIN_TOOLS = {
+    "ar": struct(bin_name = "llvm-ar", lib = "@llvm-project//llvm:llvm-ar-lib"),
+    "cgdata": struct(bin_name = "llvm-cgdata", lib = "@llvm-project//llvm:llvm-cgdata-lib"),
+    "cxxfilt": struct(bin_name = "llvm-cxxfilt", lib = "@llvm-project//llvm:llvm-cxxfilt-lib"),
+    "debuginfod-find": struct(bin_name = "llvm-debuginfod-find", lib = "@llvm-project//llvm:llvm-debuginfod-find-lib"),
+    "dsymutil": struct(bin_name = "dsymutil", lib = "@llvm-project//llvm:dsymutil-lib"),
+    "dwp": struct(bin_name = "llvm-dwp", lib = "@llvm-project//llvm:llvm-dwp-lib"),
+    "gsymutil": struct(bin_name = "llvm-gsymutil", lib = "@llvm-project//llvm:llvm-gsymutil-lib"),
+    "ifs": struct(bin_name = "llvm-ifs", lib = "@llvm-project//llvm:llvm-ifs-lib"),
+    "libtool-darwin": struct(bin_name = "llvm-libtool-darwin", lib = "@llvm-project//llvm:llvm-libtool-darwin-lib"),
+    "lipo": struct(bin_name = "llvm-lipo", lib = "@llvm-project//llvm:llvm-lipo-lib"),
+    "ml": struct(bin_name = "llvm-ml", lib = "@llvm-project//llvm:llvm-ml-lib"),
+    "mt": struct(bin_name = "llvm-mt", lib = "@llvm-project//llvm:llvm-mt-lib"),
+    "nm": struct(bin_name = "llvm-nm", lib = "@llvm-project//llvm:llvm-nm-lib"),
+    "objcopy": struct(bin_name = "llvm-objcopy", lib = "@llvm-project//llvm:llvm-objcopy-lib"),
+    "objdump": struct(bin_name = "llvm-objdump", lib = "@llvm-project//llvm:llvm-objdump-lib"),
+    "profdata": struct(bin_name = "llvm-profdata", lib = "@llvm-project//llvm:llvm-profdata-lib"),
+    "rc": struct(bin_name = "llvm-rc", lib = "@llvm-project//llvm:llvm-rc-lib"),
+    "readobj": struct(bin_name = "llvm-readobj", lib = "@llvm-project//llvm:llvm-readobj-lib"),
+    "sancov": struct(bin_name = "sancov", lib = "@llvm-project//llvm:sancov-lib"),
+    "size": struct(bin_name = "llvm-size", lib = "@llvm-project//llvm:llvm-size-lib"),
+    "symbolizer": struct(bin_name = "llvm-symbolizer", lib = "@llvm-project//llvm:llvm-symbolizer-lib"),
+}
+
+# A collection of additional alias names that should be available for the main
+# tools. The key is the main tool with the support for these names, followed by
+# a list of the aliased names.
+#
+# Note that we don't track separate binary names for the alias names as those
+# are always formed by prepending `llvm-` for the aliases.
+LLVM_TOOL_ALIASES = {
+    "ar": ["ranlib", "lib", "dlltool"],
+    #"cxxfilt": ["c++filt"],
+    "objcopy": ["bitcode-strip", "install-name-tool", "strip"],
+    "objdump": ["otool"],
+    "rc": ["windres"],
+    "readobj": ["readelf"],
+    "symbolizer": ["addr2line"],
+}
+
+_DEF_FILE_TEMPLATE = """
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This is a generated X-macro header for defining LLVM tools.
```

```suggestion
// This is a generated X-macro header for defining LLVM tools. It does not use
// `#include` guards, and instead is designed to be `#include`ed after the
// x-macro is defined in order for its inclusion to expand to the desired
// output. Macro definitions are cleaned up at the end of this file.
```

(Additional text stolen from `token_kind.def`)